src/ORM/Table.php

```diff
@@ -1951,13 +1951,6 @@ protected function _processSave(EntityInterface $entity, ArrayObject $options):
                 return false;
             }
 
-            if ($result !== false && !($result instanceof EntityInterface)) {
-                throw new DatabaseException(sprintf(
-                    'The beforeSave callback must return `false` or `EntityInterface` instance. Got `%s` instead.',
```

Comment:
This message was better than what users would get now. Could we use an assert here so we can give a better error message?